sound/soc/intel/boards/sof_sdw.c

```diff
@@ -1442,7 +1455,12 @@ static int sof_card_dai_links_create(struct device *dev,
 		cpus[cpu_id].dai_name = cpu_name;
 		init_dai_link(dev, links + link_index, be_id, name, 1, 1,
 				cpus + cpu_id, 1, ssp_components, 1, NULL, NULL);
+		INC_ID(be_id, cpu_id, link_index);
 	}
+	/* Echo reference */
+	init_dai_link(dev, links + link_index, be_id, "Echo-reference", 0, 1,
+			sof_dummy_component, 1, dummy_component, 1, NULL, NULL);
```

Comment:
links were allocated at line 1248 / 1261 to hold `num_links` number of links, here you will be over that.

Reply:
> links were allocated at line 1248 / 1261 to hold `num_links` number of links, here you will be over that.

Thanks @ujfalusi. Indeed, I should increase  `num_links` before that. I don't know how did it work on my test.